zebrad/src/components/mempool/downloads.rs

```diff
@@ -315,6 +315,27 @@ where
         Ok(())
     }
 
+    /// Cancel download/verification tasks of transactions with the
+    /// given transaction hash (see [`UnminedTxId::mined_id`]).
+    pub fn cancel(&mut self, mined_id: &transaction::Hash) {
+        // TODO: this requires going through the entire list of running tasks.
+        // If this becomes an issue, another HashMap may be needed.
```

Comment:
Optional performance nit:

As you've already noted, the loop in the caller and the iterator in this method make the overall check `O(m*n)`, where `m` is the number of transactions in the block, and `n` is the number of transactions in the mempool.

To make it `O(n)`, we could pass a `HashSet<transaction::Hash>` to this method, and check each mempool transaction against the `HashSet`.

But since `MAX_INBOUND_CONCURRENCY` is 10, this is unlikely to become an issue in practice.

Reply:
Creating a big set should be pretty fast, because it only depends on the size of the set.

So the overall cost is `O(m + n)`, which is still an improvement.
(Unless the constants are large.)

Reply:
Done in 8aaf53279